stl/inc/mutex

```diff
@@ -85,8 +95,11 @@ private:
 
 class mutex : public _Mutex_base { // class for mutual exclusion
 public:
-    /* constexpr */ mutex() noexcept // TRANSITION, ABI
-        : _Mutex_base() {}
+#ifdef __CUDACC__ // TRANSITION, CUDA
+    mutex() noexcept : _Mutex_base() {}
+#else // ^^^ TRANSITION, CUDA ^^^ / vvv not CUDA vvv
+    constexpr mutex() noexcept : _Mutex_base() {}
+#endif // ^^^ not CUDA ^^^

```

Comment:
Should we use `= default` here (leaving the constexpr property to be deduced) to reduce conditional compilation?